tests/dummy/app/controllers/index.js

```diff
@@ -33,6 +35,38 @@ export default Ember.Controller.extend({
 
     stopAddingContent() {
       window.clearInterval(this.intervarTimer);
+    },
+
+    triggerMouseEnter(dropdown /*, e */) {
+      if (this.closeTimer) {
+        run.cancel(this.closeTimer);
+        this.closeTimer = null;
+      } else {
+        dropdown.actions.open();
+      }
+    },
+
+    triggerMouseLeave(dropdown /*, e */) {
+      this.closeTimer = run.next(this, function() {
```

Comment:
you should likely also cancel any current `closeTimer` before setting a new one, that way the system cannot accidentally orphan a timer.

Reply:
@stefanpenner How did you come across this code that doesn't even exist today and was on the dummy app?

Reply:
I was showing @stefanpenner some old PRs I did on ember-basic-dropdown and the eventual solution you landed on for hover support.

`3. Miguel’s eventual approach: https://github.com/cibernox/ember-basic-dropdown/pull/195/files`

Nothing actionable here 😅

Reply:
All right Sir. Carry on! 😄